src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx

```diff
@@ -80,24 +81,28 @@ const CreateOrEditQuizContainer = ({
   }
 
   const handleResponse = () => {
-    setAlert({
-      severity: snackbarVariants.success,
-      message: id
-        ? t('myResourcesPage.quizzes.successEditedQuiz')
-        : t('myResourcesPage.quizzes.successAddedQuiz')
-    })
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.success,
+        message: id
+          ? t('myResourcesPage.quizzes.successEditedQuiz')
+          : t('myResourcesPage.quizzes.successAddedQuiz')
+      })
+    )
     navigateToQuizzesTab()
   }
 
   const onResponseError = (error: ErrorResponse) => {
-    setAlert({
-      severity: snackbarVariants.error,
-      message: error
-        ? t(`errors.${error.code}`, {
-            message: getErrorMessage(error.message)
-          })
-        : ''
-    })
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.error,
+        message: error
+          ? t(`errors.${error.code}`, {
```

Comment:
Isn't here will be some kind of error, since you already included `t` in snack bar component?

Reply:
Thanks for noticing. There won't be error actually. In other places, I removed `t` from `openAlert` call, but here we also pass some options, not just string, so I kept it before. But it really looks unnecessary, to use `t` here and inside of `AppSnackbar`, so I added possibility to pass translation options directly to snackbar redux state. So now snackbar message type looks like this 
```
interface ExtendedSnackbarMessage {
  text: string
  options: WritableDraft<TOptions>
}
type SnackbarMessage = string | ExtendedSnackbarMessage
```
`WritableDraft` is used because some of TOptions properties use `readonly`, so ts throws error without it. Besides, we don't change those properties directly.

Reply:
Nice solution!